src/main/java/org/verdictdb/coordinator/SelectQueryCoordinator.java

```diff
@@ -169,6 +173,64 @@ public void abort() {
     planRunner.abort();
   }
 
+  /**
+   * @param query Select query
+   * @return check if the query contain the syntax that is not supported by VerdictDB
+   */
+  private void ensureQuerySupport(SelectQuery query) throws VerdictDBException {
+    // current, only if count distinct appear along with other aggregation function will return false
+
+    // check select list
+    boolean containAggregatedItem = false;
+    boolean containCountDistinctItem = false;
+    for (SelectItem selectItem:query.getSelectList()) {
+      if (selectItem instanceof AliasedColumn &&
+          ((AliasedColumn) selectItem).getColumn() instanceof ColumnOp) {
+        if (((ColumnOp) ((AliasedColumn) selectItem).getColumn()).doesColumnOpContainOpType("countdistinct")) {
+          ((ColumnOp) ((AliasedColumn) selectItem).getColumn()).replaceAllColumnOpOpType("countdistinct", "approx_countdistinct");
+          containCountDistinctItem = true;
+        }
+        if (!containAggregatedItem &&
+            (((AliasedColumn) selectItem).getColumn()).isAggregateColumn()) {
+          containAggregatedItem = true;
+        }
+        if (containAggregatedItem && containCountDistinctItem) {
+          throw new VerdictDBException("Count distinct and other aggregation should not both appear in one query");
+        }
+      }
+    }
+    // check from list
+    for (AbstractRelation table:query.getFromList()) {
+      if (table instanceof SelectQuery) {
+        ensureQuerySupport((SelectQuery) table);
+      }
+      else if (table instanceof JoinTable) {
+        for (AbstractRelation jointable:((JoinTable) table).getJoinList()) {
+          if (jointable instanceof SelectQuery) {
+            ensureQuerySupport((SelectQuery) jointable);
+          }
+        }
+      }
+    }
+
+    // also need to check having since we will convert having clause into select list
+    if (query.getHaving().isPresent()) {
+      UnnamedColumn having = query.getHaving().get();
+      if (having instanceof ColumnOp &&
+          ((ColumnOp)having).doesColumnOpContainOpType("countdistinct")) {
+        containCountDistinctItem = true;
+        ((ColumnOp) having).replaceAllColumnOpOpType("countdistnct", "approx_countdistinct");
+      }
+      if (having instanceof ColumnOp &&
+          having.isAggregateColumn()) {
+        containAggregatedItem = true;
+      }
+      if (containAggregatedItem && containCountDistinctItem) {
+        throw new VerdictDBException("Count distinct and other aggregation should not both appear in one query");
```

Comment:
They can appear in two different select lists, right? Like one in an outer query and the other in an inner query. Otherwise, this check should be modified.

Also, please change the error message to "Count distinct and other aggregate functions cannot appear in the same select list."

Reply:
What is "this situation"?

Reply:
The situation that aggregate item and count distinct are in the selectlist of same query

Reply:
My question was if a count-distinct function and another regular aggregate function (e.g., sum) can appear in two different select lists in the same query.

Reply:
I see. I think redshift query 13 is the case you mean. 
```
SELECT COUNT(DISTINCT "customer"."c_custkey") AS "ctd_c_custkey_ok",
  "t0"."__measure__0" AS "calculation_1361031105746580"
FROM "public"."customer" "customer"
  INNER JOIN (
  SELECT COUNT((CASE WHEN ((0 < STRPOS("orders"."o_comment",CAST('special' AS TEXT))) AND (0 < (CASE
  	WHEN STRPOS("orders"."o_comment",CAST('special' AS TEXT)) IS NULL THEN NULL
  	WHEN 0 = STRPOS(SUBSTRING("orders"."o_comment",INT4LARGER(1,CAST(FLOOR(STRPOS("orders"."o_comment",CAST('special' AS TEXT))) AS INTEGER))),CAST('requests' AS TEXT)) THEN 0
  	ELSE (STRPOS(SUBSTRING("orders"."o_comment",INT4LARGER(1,CAST(FLOOR(STRPOS("orders"."o_comment",CAST('special' AS TEXT)))AS INTEGER))),CAST('requests' AS TEXT)) + INT4LARGER(1,CAST(FLOOR(STRPOS("orders"."o_comment",CAST('special' AS TEXT))) AS INTEGER)) - 1)	END))) THEN NULL ELSE "orders"."o_orderkey" END)) AS "__measure__0",
    "customer"."c_custkey" AS "c_custkey"
  FROM "public"."orders" "orders"
    RIGHT JOIN "public"."customer" "customer" ON ("orders"."o_custkey" = "customer"."c_custkey")
  GROUP BY 2
) "t0" ON (("customer"."c_custkey" = "t0"."c_custkey") OR (("customer"."c_custkey" IS NULL) AND ("t0"."c_custkey" IS NULL)))
GROUP BY 2
ORDER BY 2
```
I think as long as the asyncNode of inner query performs right, the accuracy will be gradually improved with increasing block number.

Reply:
I know that.  So does the function you implemented work as follows?
1. Pass if a count-distinct function and another regular aggregate function (e.g., sum) appear in **two different select lists in the same query**
2. Throws an error if a count-distinct function and another regular aggregate function (e.g., sum) appear **in the same select list.**

Reply:
The current logic is checking the select list (also having) to find whether each item contains either aggregate item or count distinct.  If select list contain both it will throw an error. 

And it will do the same process recursively in the from list.

Reply:
Thanks for the explanation, but is the answer to my question yes or no? See that my question starts with "does", not "how".

Reply:
Yes, it should have. Additionally, the case like
```
select some_column, count(*) 
from table 
group by some_column 
having count(distinct some_column)
```
will fail.

Reply:
I see. Would it be hard to place this ensureQuerySupport function after the process "having expression -> select item" is performed? Then, this logic itself does not have to explicitly check the having clause, which will reduce redundancy. If the conversion process is deep inside the query processing logic, the current approach is fine for now.

FYI, "it should have" means "it should" but "it didn't".

Reply:
Sorry for the poor syntax. I think it is doable. The having converting process is in QueryExecutionPlanFactory.createAggExecutionAndItsDependents() method. So we can place the checking method in SelectQueryCoordinator right after QueryExecutionPlan is created(). We can track source to find the subqueries of node and recursively check the subqueries.

Reply:
I see, but let's keep the current stage then. I think the supported query checking should be performed before converting a query to execution nodes.

Please just change the error message and push. I'll merge this after.